__tests__/__utils__/kv.ts

```diff
@@ -30,11 +30,9 @@ export function createKV<Key extends string>(): KV<Key> {
     // eslint-disable-next-line @typescript-eslint/require-await
     async put(key: Key, value: string, _?: { expirationTtl: number }) {
       values[key] = value
-      expect(key.length).toBeLessThanOrEqual(512)
+      if (key.length > 512) {
+        throw 'Error: key longer than 512 characters.'
+      }
```

Comment:
Right! This check should be added before the value is saved, e.g. before the line `values[key] = value`. BTW: `expect(key)...` would work here -> However it is not a good stil. Here we simulate the real behavior of a KV store and a KV store would raise an error.

Reply:
ok! should I add a else to the if? I mean, if it throw the error does it still save the value or not?

Reply:
No, that's not necessary since with the throw statement the execution is stopped. Thus

```ts
if (key.length > 512) {
  throw 'Error: key longer than 512 characters.'
}
values[key] = value
```

is the same as

```ts
if (key.length > 512) {
  throw 'Error: key longer than 512 characters.'
} else {
  values[key] = value
}
```

It is common practice to not having an else statement when the if statement is a check which stops the execution. Thus we save an indention in the else block.